credential-controller.js

```diff
@@ -109,6 +109,7 @@ function factory(
 
   function init() {
     brRefreshService.register($scope, function(force) {
```

Comment:
It seems odd to me that this is all happening inside of a refresh function. If the user is in the middle of their flow and a refresh event occurs ... and their session hasn't changed or anything of that nature, can the UI suddenly change the display and unnecessarily disrupt them? We might want to be more careful with what happens on refresh and move most of this code elsewhere.